yt_dlp/extractor/applepodcasts.py

```diff
@@ -39,47 +51,25 @@ class ApplePodcastsIE(InfoExtractor):
     def _real_extract(self, url):
         episode_id = self._match_id(url)
         webpage = self._download_webpage(url, episode_id)
-        episode_data = {}
-        ember_data = {}
-        # new page type 2021-11
-        amp_data = self._parse_json(self._search_regex(
-            r'(?s)id="shoebox-media-api-cache-amp-podcasts"[^>]*>\s*({.+?})\s*<',
-            webpage, 'AMP data', default='{}'), episode_id, fatal=False) or {}
-        amp_data = try_get(amp_data,
-                           lambda a: self._parse_json(
-                               next(a[x] for x in iter(a) if episode_id in x),
-                               episode_id),
-                           dict) or {}
-        amp_data = amp_data.get('d') or []
-        episode_data = try_get(
-            amp_data,
-            lambda a: next(x for x in a
-                           if x['type'] == 'podcast-episodes' and x['id'] == episode_id),
-            dict)
-        if not episode_data:
-            # try pre 2021-11 page type: TODO: consider deleting if no longer used
-            ember_data = self._parse_json(self._search_regex(
-                r'(?s)id="shoebox-ember-data-store"[^>]*>\s*({.+?})\s*<',
-                webpage, 'ember data'), episode_id) or {}
-            ember_data = ember_data.get(episode_id) or ember_data
-            episode_data = try_get(ember_data, lambda x: x['data'], dict)
-        episode = episode_data['attributes']
-        description = episode.get('description') or {}
-
-        series = None
-        for inc in (amp_data or ember_data.get('included') or []):
-            if inc.get('type') == 'media/podcast':
-                series = try_get(inc, lambda x: x['attributes']['name'])
-        series = series or clean_html(get_element_by_class('podcast-header__identity', webpage))
+        server_data = self._parse_json(
+            get_element_by_id('serialized-server-data', webpage),
+            episode_id) or [{}]
+        model_data = traverse_obj(
+            server_data,
+            (0, 'data', 'headerButtonItems',
+             {lambda x: next(y for y in x
+                             if y.get('$kind') == 'bookmark' and y.get('modelType') == 'EpisodeOffer')},
+             'model'))
+        schema_content = traverse_obj(server_data, (0, 'data', 'seoData', 'schemaContent'))
```

Comment:
Instead of parsing the json-ld schema ourselves, we can use the already existing helper functions
```suggestion
```

Reply:
ah, yes, use `_yield_json_ld()` to process the `<script id=schema:episode type="application/ld+json">` in the webpage and automatically populate attributes instead of doing it manually on the copy of it embedded in `<script type="application/json" id="serialized-server-data">`.  TIL!

Reply:
You could normally use `self._search_json_ld` to both search for the json-ld block in the HTML and parse it. But in this case, I would split it up as shown in the suggestion below to use both the data from the hydration data in `schemaContent` and the json-ld block in the HTML as a fallback.

Reply:
yes, searching would just find and use `<script id=schema:episode type="application/ld+json">`, but the way you suggested allows for two sources of the schema (for redundancy).